R/dependencies.R

```diff
@@ -825,3 +825,85 @@ update_with_direct_deps <- function(project = ".",
   )
 }
 
+#' Recursively discover and build, check and install
+#' internal dependencies
+#'
+#' It discovers all dependencies starting from the repositories,
+#' determines the installation order and then
+#' builds, checks and installs them in order.
+#'
+#' @md
+#' @inheritParams rec_checkout_internal_deps
+#' @param artifact_dir (`character`) directory where build
+#'   tarball and logs go to
+#' @export
+#' @examples
+#' \dontrun{
+#' build_check_install_repos(
+#'   list(list(repo = "insightsengineering/teal", host = "https://github.com")),
+#'   feature = "main",
+#'   direction = "upstream",
+#'   local_repos = data.frame(repo = "insightsengineering/teal",
+#'   host = "https://github.com", directory = "../scratch1/teal/",
+#'   stringsAsFactors = FALSE)
+#' )
+#' build_check_install_repos(
+#'   list(list(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   host = "https://code.roche.com")),
+#'   feature = "master",
+#'   direction = "upstream",
+#'   #local_repos = data.frame(repo = "maximilian_oliver.mordig/stageddeps.electricity",
+#'   #host = "https://code.roche.com",
+#'   #directory = "../example_ecosystem/stageddeps.electricity/",
+#'   #stringsAsFactors = FALSE),
```

Comment:
don't think we should have commented out args in the example

Reply:
Did you mean to uncomment the `local_repos` arg? I did so now.

Reply:
Did you mean to uncomment the `local_repos` arg? I did so now.